### PR TITLE
Use Heroku's Release Phase for Migrations

### PR DESCRIPTION
**Pull request by mcastle**
Automatically run migrations on deployments to Heroku. Advantages include deployments are rolled-back if a migration fails, preventing broken applications due to failed migrations, no time between when application is released and database is migrated, and removes risk of forgetting to manually run migrations.

### CODE REVIEW

**Comment by mcastle** [comment]
I don't think the Heroku deployment documentation needs to account for this, since the new code is only in the Procfile and doesn't require any extra configuration or Heroku commands to be run (it runs automatically after a typical git push to Heroku).

**Comment by browniebroke** [comment]
Sorry, I wasn't clear, what I meant is that the documentation can be simplified, we don't need to run `heroku python manage.py migrate` anymore, therefore, we can remove that line from the documentation.

**Comment by mcastle** [comment]
No prob, I should've caught that. Done! https://github.com/pydanny/cookiecutter-django/pull/1630

File: {{cookiecutter.project_slug}}/Procfile [hunk 1]
```diff
@@ -1,3 +1,4 @@
+release: ./manage.py migrate
```
Comment (webyneter) [inline {{cookiecutter.project_slug}}/Procfile]: Let's make it `release: manage.py migrate`, shall we? It's pretty much obvious what `manage.py` we refer to here anyway.

**Comment by browniebroke** [comment]
FYI, we had to revert this, see #1646 for more details.

**Pull request merged**